hyperspy/tests/utils/test_roi.py

```diff
@@ -486,6 +486,37 @@ def test_line2droi_angle(self):
         with pytest.raises(ValueError):
             r.angle(axis='z')
 
+    def test_repr_None(self):
+        # Setting the args=None sets them as traits.Undefined, which didn't 
+        # have a string representation in the old %s style.
```

Comment:
Rewrite this to something along the line of:
```python
    for roi in [Point1DROI, Point2DROI, RectangularROI, SpanROI]:
        r = roi()
        for value in tuple(r):
            assert value == t.Undefined
        repr(r)
    for roi in [CircleROI, Line2DROI]:
        r = roi()
        for value in tuple(r)[:-2]:
            assert value == t.Undefined
        assert tuple(r)[-1] == 0
        repr(r)
```

Reply:
Great. Used your example, but changed [::-2] to [::-1], as only the final parameter is != Undefined.